scripts/cmake/CompilerSetup.cmake

```diff
@@ -36,129 +30,98 @@ if(BUILD_SHARED_LIBS)
     set(CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS TRUE)
 endif()
 
-if(OGS_ENABLE_AVX2)
-    set(CPU_FLAGS "-mavx2 -march=core-avx2")
-elseif(OGS_CPU_ARCHITECTURE STREQUAL "generic")
-    set(CPU_FLAGS "-mtune=generic")
-else()
-    set(CPU_FLAGS "-march=${OGS_CPU_ARCHITECTURE}")
+### GNU-like compiler
+if(COMPILER_IS_GCC OR COMPILER_IS_CLANG OR COMPILER_IS_INTEL)
+    if(NOT CMAKE_BUILD_TYPE STREQUAL "Debug")
+        message(STATUS "Set release compiler flags")
+        add_compile_options(-O3)
+    elseif(NOT STL_NO_DEBUG)
+        # Enable assertions in STL in debug mode.
```

Comment:
No they are not defined. They add debug symbols to the libc++. This was added by @endJunction. I checked the [docs](http://gcc.gnu.org/onlinedocs/libstdc++/manual/using_macros.html) and found only `_GLIBCXX_DEBUG` and `_GLIBCXX_DEBUG_PEDANTIC`. I am not sure if the other defines are legit.

Reply:
Thanks for checking the docs. I thought, bounds checking was enabled by default in debug mode -- another misconception clarified.

Reply:
I guess they are dependent on the compiler version. The linked docs are for the newest version. I think it would be not wrong to add `_GLIBCXX_DEBUG_PEDANTIC` to the list.